packages/storage/src/indexed-db/index.ts

```diff
@@ -802,4 +795,44 @@ export class IndexedDb implements IndexedDbInterface {
       output: Value.fromJson(result.output),
     };
   }
+
+  async fetchStakingTokenId(): Promise<AssetId> {
+    const registryClient = new ChainRegistryClient();
+    const registry = registryClient.get(this.chainId);
+    const stakingTokenMetadata = registry.getMetadata(registry.stakingAssetId);
+    const stakingToken = getAssetId(stakingTokenMetadata);
+
+    return stakingToken;
+  }
+
+  async hasStakingAssetBalance(assetId: AssetId): Promise<boolean> {
+    const spendableUMNotes = await this.db.getAllFromIndex(
+      'SPENDABLE_NOTES',
+      'assetId',
+      uint8ArrayToBase64(assetId.inner),
+    );
+
+    // Iterate over the spendable UM notes, and acrue balance for unspent notesz
+    let stakingTokenBalance = new Amount();
+    for (const note of spendableUMNotes) {
+      let umNote = note as unknown as SpendableNoteRecord;
+      if (umNote.heightSpent == undefined) {
+        const newAmount = addLoHi(
+          { lo: stakingTokenBalance.lo, hi: stakingTokenBalance.hi },
+          {
+            lo: BigInt(umNote.note?.value?.amount?.lo ?? 0n),
+            hi: BigInt(umNote.note?.value?.amount?.hi ?? 0n),
+          },
+        );
+        stakingTokenBalance.lo = newAmount.lo;
+        stakingTokenBalance.hi = newAmount.hi;
+      }
```

Comment:
Couldn't this all be simplified a bit? e.g., 

```suggestion
        stakingTokenBalance = addLoHi(
          stakingTokenBalance,
          {
            lo: BigInt(umNote.note?.value?.amount?.lo ?? 0n),
            hi: BigInt(umNote.note?.value?.amount?.hi ?? 0n),
          },
        );
      }
```

Reply:
this doesn't seem to work because of a type signature mismatch, ie. you're trying to assign `Required<LoHi>` to an `Amount`. I"ll make some changes to the variable names here.

Reply:
Ahh... I missed that `hi` was optional on a `LoHi`, and I'd assumed that `PartialMessage<Amount>` and `LoHi` were interchangeable. My mistake.